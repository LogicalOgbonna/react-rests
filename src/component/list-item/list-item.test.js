import React from "react";
import { shallow } from 'enzyme';
import { checkProps, findByTestAttr, renderWithoutCrashing } from '../../utils/tests';
import ListItem from "./list-item";
const expectedProps = {
    title: "soe",
    description: "descrip"
};

describe('<ListItem /> Component', () => {
    describe('Checking Proptypes', () => {
        it('should not throw a warning', () => {
            const errProps = checkProps(ListItem, expectedProps);
            expect(errProps).toBeUndefined();
        })

    })

    describe('Render <ListItem />', () => {
        let component;
        beforeEach(() => {
            component = shallow(<ListItem {...expectedProps} />)
        });
        renderWithoutCrashing(component, "<ListItem />");
        it('should Render title in <ListItem />', () => {
            const listItem = findByTestAttr(component, "title");
            expect(listItem.length).toBe(1)
        })
        it('should Render description in <ListItem />', () => {
            // console.log()
            const listItem = findByTestAttr(component, "description");
            console.log(listItem.props())
            // expect(listItem.props().title).toEqual("soe")
            expect(listItem.length).toBe(1)
        })
    })

    describe('Should Return null', () => {
        let component;
        beforeEach(() => {
            component = shallow(<ListItem />);
        });

        it('should return null if no title passed to <ListItem />', () => {
            const listItem = findByTestAttr(component, "listItemComponent");
            expect(listItem.length).toBe(0)
        })
    })


})
