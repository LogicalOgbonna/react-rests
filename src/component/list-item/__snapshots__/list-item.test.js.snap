// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> Component Render <ListItem /> should render <ListItem /> Without Crashing 1`] = `undefined`;
